Decrypt obm settings when testing OBM configs in create-X-obm-setting…

benbp opened the pull request:
…s jobs

When we migrated OBM decryption over to the model layer, we forgot about this piece of code.

Requires https://github.com/RackHD/on-core/pull/192

Edit: updated for @yyscamper's comments.

@RackHD/corecommitters @hohene @johren

heckj commented:
👍

benbp commented:
test this please

yyscamper commented on spec/lib/jobs/create-obm-settings-spec.js:
```diff
@@ -65,12 +69,45 @@ describe(require('path').basename(__filename), function () {
+            var jobOptions = {
+                service: 'ipmi-obm-service',
+                config: {
+                    alias: 'test',
```
Why add `alias` here? I think this may confuse user if user reads the unit-test to study how to use this function.

benbp replied:
Whoops!

yyscamper commented:
I'd support move the decryption to on-core's model, I am in favor to hide the encryption & decryption in model level and being transparent to job and other modules.

Meanwhile, I checked the function `liveTestIpmiConfig`, it created ObmService with input obmConfig, but if we use this function [waterline.obms.findByNode](https://github.com/RackHD/on-core/blob/master/lib/models/obms.js#L123) with the `reveal` set to true, we can not only get ride of the helpers.revealSecret but also verify whether the obm setting is upserted correctly.

benbp commented:
Makes sense @yyscamper. I'll do a re-work today.

benbp commented:
@yyscamper updated, I added a reveal option to the `models.obm.upsertByNode` method (instead of findByNode, this way we avoid an extra database call) in https://github.com/RackHD/on-core/pull/192, and also removed the now unused revealSecrets function in the job-helper module.

yyscamper commented:
@benbp : It's good for me, :+1:
